spec/rest-middleware.spec.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+describe("restMiddleware", () => {
+    var configMock, helmetMock, logger, restMiddleware, serverMock, restifyLinks;
+
+    beforeEach(() => {
+        var LoggerMock, RestMiddleware;
+
+        RestMiddleware = require("../lib/rest-middleware");
+        LoggerMock = require("./mock/logger-mock");
+        helmetMock = jasmine.createSpyObj("helmetMock", [
+            "frameguard",
+            "hidePoweredBy",
+            "hsts",
+            "ieNoOpen",
+            "noCache",
+            "noSniff",
+            "xssFilter"
+        ]);
+        serverMock = jasmine.createSpyObj("serverMock", [
+            "use"
+        ]);
+        configMock = {
+            baseUrl: "http://localhost:8443",
+            https: false
+        };
+        restifyLinks = jasmine.createSpy();
+        logger = new LoggerMock();
+        restMiddleware = new RestMiddleware(helmetMock, logger, restifyLinks);
+    });
+
+    function expectNormalMiddlewareWasCalled () {
+        expect(helmetMock.frameguard).toHaveBeenCalled();
+        expect(helmetMock.ieNoOpen).toHaveBeenCalled();
+        expect(helmetMock.hidePoweredBy).toHaveBeenCalled();
+        expect(helmetMock.ieNoOpen).toHaveBeenCalled();
+        expect(helmetMock.noCache).toHaveBeenCalled();
+        expect(helmetMock.noSniff).toHaveBeenCalled();
+        expect(helmetMock.xssFilter).toHaveBeenCalled();
+        expect(restifyLinks).toHaveBeenCalled();
+    }
+
+    it("calls restMiddleware without https", () => {
+        restMiddleware(configMock, serverMock);
```

Comment:
To elaborate my previous point, I would change this line to read

    restMiddleware({}, serverMock);